lib/fetcher/v1_alpha_fetcher.rb

```diff
@@ -0,0 +1,47 @@
+require_relative "fetcher"
+
+class V1AlphaFetcher < Fetcher
+  def fetch(base_url)
+    base_url += "/apis"
+
+    uri = URI(base_url)
+
+    req = Net::HTTP::Get.new(uri)
+    req["correlation-id"] = SecureRandom.uuid
+    req["accept"] = "application/vnd.uk.gov.api.v1alpha+json"
+
+    response = Net::HTTP.start(uri.hostname, uri.port, use_ssl: uri.scheme == "https") do |http|
+      http.request(req)
+    end
+    raise VersionNotSupportedError, "Could not parse APIs from #{base_url}, no version supported" if response.is_a? Net::HTTPNotAcceptable
+
+    raise VersionNotSupportedError, "A client error caused retrieval of APIs from #{base_url} to fail" if response.is_a? Net::HTTPNotFound
+
+    response = JSON.parse response.body
+
+    apis = response["apis"].map do |api|
+      data = api["data"]
+      Api.new name: data["name"],
+              description: data["description"],
+              url: data["url"],
+              documentation: data["documentation-url"],
+              provider: data["organisation"].gsub(/\s+/, "-").downcase
```

Comment:
Although only 2 duplications so far, I wonder if we should extract  `data["organisation"].gsub(/\s+/, "-").downcase` to a method